big_tests/tests/sasl_external_SUITE.erl

```diff
@@ -64,138 +85,144 @@ end_per_suite(Config) ->
+                generate_user_tcp(C, User)],
+    cert_fails_to_authenticate(UserSpec).
 
-cert_with_cn_no_xmpp_addrs_requested_correct_user(C) ->
-    UserSpec = [{requested_name, <<"john@localhost">>} |
-		generate_user_tcp(C, "john")],
+cert_no_xmpp_addrs_just_use_identity(C) ->
+    User = username("not-mike", C),
+    UserSpec = [{requested_name, <<"mike@localhost">>} |
+		generate_user_tcp(C, User)],
     {ok, Client, _} = escalus_connection:start(UserSpec),
-
     escalus_connection:stop(Client).
 
-cert_with_cn_xmpp_addrs_request_name_empty(C) ->
-    UserSpec = generate_user_tcp(C, "not-alice-name"),
+cert_no_xmpp_addrs_no_identity(C) ->
+    User = username("john", C),
+    UserSpec = generate_user_tcp(C, User),
     cert_fails_to_authenticate(UserSpec).
 
-cert_with_cn_one_xmpp_addrs_request_name_empty(C) ->
-    UserSpec = generate_user_tcp(C, "bob"),
-    {ok, Client, _} = escalus_connection:start(UserSpec),
+cert_more_xmpp_addrs_no_identity_fails(C) ->
+    User = username("not-alice", C),
+    UserSpec = generate_user_tcp(C, User),
+    cert_fails_to_authenticate(UserSpec).
 
+cert_one_xmpp_addrs_no_identity(C) ->
+    User = username("bob", C),
+    UserSpec = generate_user_tcp(C, User),
+    {ok, Client, _} = escalus_connection:start(UserSpec),
     escalus_connection:stop(Client).
 
-cert_with_cn_no_xmpp_addrs_request_name_empty(C) ->
-    UserSpec = generate_user_tcp(C, "john"),
+cert_with_cn_no_xmpp_addrs_no_identity(C) ->
+    User = username("john", C),
+    UserSpec = generate_user_tcp(C, User),
     {ok, Client, _} = escalus_connection:start(UserSpec),
-
     escalus_connection:stop(Client).
 
-cert_no_cn_no_xmpp_addrs_request_name_empty(C) ->
-    UserSpec = generate_user_tcp(C, "john"),
-    cert_fails_to_authenticate(UserSpec).
+cert_with_cn_no_xmpp_addrs_identity_correct(C) ->
+    User = username("john", C),
+    UserSpec = [{requested_name, requested_name(User)} |
+                generate_user_tcp(C, User)],
+    {ok, Client, _} = escalus_connection:start(UserSpec),
+    escalus_connection:stop(Client).
 
-cert_with_cn_no_xmpp_addrs_request_wrong_name(C) ->
+cert_with_cn_no_xmpp_addrs_wrong_identity_fails(C) ->
+    User = username("not-mike", C),
     UserSpec = [{requested_name, <<"mike@localhost">>} |
-		generate_user_tcp(C, "not-mike-name")],
+                generate_user_tcp(C, User)],
     cert_fails_to_authenticate(UserSpec).
 
-cert_with_cn_xmpp_addrs_request_wrong_name(C) ->
-    UserSpec = [{requested_name, <<"grace@localhost">>} |
-		generate_user_tcp(C, "grace-no-address")],
+cert_more_xmpp_addrs_wrong_identity_fails(C) ->
+    User = username("grace", C),
+    UserSpec = [{requested_name, requested_name(User)} |
+                generate_user_tcp(C, User)],
     cert_fails_to_authenticate(UserSpec).
 
-cert_with_cn_one_xmpp_addr_requested_wrong_hostname(C) ->
-    UserSpec = [{requested_name, <<"bob@fed1">>} |
-		generate_user_tcp(C, "bob")],
+cert_one_xmpp_addr_wrong_hostname(C) ->
```

Comment:
This is just "bob@loclhost", so the test case is the same as `cert_one_xmpp_addr_identity_correct`.